modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java

```diff
@@ -612,6 +619,144 @@ public void testAddObjectEntry() throws Exception {
 				).build()));
 	}
 
+	@Test
+	public void testAddObjectEntryWithEncryptedObjectField() throws Exception {
+		ObjectFieldTestUtil.testWithEncryptedObjectFieldProperties(
+			"AES", true, ObjectFieldTestUtil.generateKey("AES"),
+			() -> {
+				_addCustomObjectField(
+					new EncryptedObjectFieldBuilder(
+					).externalReferenceCode(
+						"encryptedFieldErc"
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap(
+							RandomTestUtil.randomString())
+					).name(
+						"encrypted"
+					).objectDefinitionId(
+						_objectDefinition.getObjectDefinitionId()
+					).build());
+
+				ObjectEntry objectEntry = _addObjectEntry(
+					HashMapBuilder.<String, Serializable>put(
+						"emailAddress", RandomTestUtil.randomString()
+					).put(
+						"emailAddressRequired", "athanasius@liferay.com"
+					).put(
+						"encrypted", "test"
+					).put(
+						"externalReferenceCode", "objectEntryErc"
+					).put(
+						"listTypeEntryKeyRequired", "listTypeEntryKey1"
+					).build());
+
+				_assertCount(1);
+
+				Assert.assertEquals(
+					"test",
+					MapUtil.getString(
+						_objectEntryLocalService.getValues(
+							objectEntry.getObjectEntryId()),
+						"encrypted"));
```

Comment:
After this assert, I would like you try to get the encrypted value inside Dynamic ObjectEntry table, peharps through a DSL query, and assert it with encrypted value of the plain text that you assign in this field when execute the `_addObjectEntry`:

```java
Assert.assertEquals(
	_encryptor.encrypt((key, "test"),
         dslQueryResult... // value "test" exactly how it is stored in database, without pass by the decrypt process
)
```